parser/ast.go

```diff
@@ -1770,18 +1770,19 @@ func (c *CreateDatabase) Accept(visitor ASTVisitor) error {
 }
 
 type CreateTable struct {
-	CreatePos    Pos // position of CREATE|ATTACH keyword
-	StatementEnd Pos
-	OrReplace    bool
-	Name         *TableIdentifier
-	IfNotExists  bool
-	UUID         *UUID
-	OnCluster    *ClusterClause
-	TableSchema  *TableSchemaClause
-	Engine       *EngineExpr
-	SubQuery     *SubQuery
-	HasTemporary bool
-	Comment      *StringLiteral
+	CreatePos       Pos // position of CREATE|ATTACH keyword
+	StatementEnd    Pos
+	OrReplace       bool
+	Name            *TableIdentifier
+	IfNotExists     bool
+	UUID            *UUID
+	OnCluster       *ClusterClause
+	TableSchema     *TableSchemaClause
+	Engine          *EngineExpr
+	SubQuery        *SubQuery
+	AsTableFunction *TableFunctionExpr
```

Comment:
What about renaming `AsTableFunction` to `TableFunction` to keep consistent with `SubQuery` which also omits the `As` prefix.